firmware/fsm.c

```diff
@@ -1637,6 +1638,280 @@ void fsm_msgCosiSign(CosiSign *msg)
 	layoutHome();
 }
 
+void fsm_msgStellarGetPublicKey(StellarGetPublicKey *msg)
+{
+    RESP_INIT(StellarPublicKey);
+
+    CHECK_INITIALIZED
+
+    CHECK_PIN
+
+    // Will exit if the user does not confirm
+    stellar_layoutGetPublicKey(msg->address_n, msg->address_n_count);
+
+    // Read public key and write it to the response
+    resp->has_public_key = true;
+    resp->public_key.size = 32;
+    stellar_getPubkeyAtAddress(msg->address_n, msg->address_n_count, resp->public_key.bytes, sizeof(resp->public_key.bytes));
+
+    msg_write(MessageType_MessageType_StellarPublicKey, resp);
+
+    layoutHome();
+}
+
+void fsm_msgStellarSignMessage(StellarSignMessage *msg)
+{
+    CHECK_INITIALIZED
+    CHECK_PIN
+
+    RESP_INIT(StellarMessageSignature);
+
+    // Will exit if the user does not confirm
+    stellar_confirmSignString(msg, resp);
+
+    msg_write(MessageType_MessageType_StellarMessageSignature, resp);
+
+    layoutHome();
+}
+
+void fsm_msgStellarVerifyMessage(StellarVerifyMessage *msg)
+{
+    if (!stellar_verifySignature(msg)) {
+        fsm_sendFailure(FailureType_Failure_DataError, _("Invalid signature"));
+        return;
+    }
+
+    fsm_sendSuccess(_("Message verified"));
+    layoutHome();
+}
+
+void fsm_msgStellarSignTx(StellarSignTx *msg)
+{
+    CHECK_INITIALIZED
+    CHECK_PIN
+
+    stellar_signingInit(msg);
+
+    // Confirm transaction basics
+    stellar_layoutTransactionSummary(msg);
```

Comment:
Like in https://github.com/trezor/trezor-mcu/pull/351#discussion_r224993593, the behavior here is a little strange if the user cancels, since both the Failure::ActionCancelled and the StellarTxOpRequest get returned to the host.